### DISCUSSION TITLE
Extend default message actions on Android

### DISCUSSION THREAD

**Discussion by NikaKurdadze** [GetStream/stream-chat-android#3517] [Q&A]
Hi, how can I add an action to the message actions list on Android? This is achieved via the `supportedMessageActions` method in the `ViewFactory` according to the [docs](https://getstream.io/chat/docs/sdk/ios/swiftui/components/message-reactions/#customizing-the-reactions-overlay-view)

**Comment by filbabic** [comment]
Hey @NikaKurdadze The docs and examples you provided are iOS related. Do you want to achieve this in Android to mimic iOS?

In the `MessageListView` we have a style that allows you to change which items are enabled/disabled, so you can remove some of the default actions. You can't currently provide custom actions in XML (UI Components) but you can in Jetpack Compose (Compose UI Components).

Also, you have access to `MessageListView.setMessageLongClickListener()` in XML if you want to override what happens when you long tap on messages. Using this, you can show your own custom UI that lets you trigger custom actions if you have them.

**Comment by NikaKurdadze** [comment]
Yes, I am trying to achieve the same behavior as the iOS doc I referenced. Is this a functionality that could be added to UI Components in the nearest future?

**Reply by filbabic** [reply]
Hey @NikaKurdadze!

I'll create a ticket for this feature request. We're pretty packed in terms of work for the next few sprints and I cannot guarantee a timeline when it would be available. It's also quite a bit of effort to include this and make it work with custom actions, which is why I suggest you try to override the long tap listener to provide custom UI and handlers when long tapping on messages as this would be a workaround for you and you'd be able to 100% control what happens with each action.

You'll be able to track our progress on [the ticket here](https://github.com/GetStream/stream-chat-android/issues/3519) and we'll see what we can do in the short term!

**Reply by NikaKurdadze** [reply]
I see. Is there any way I can reuse the existing actions dialog?

**Reply by filbabic** [reply]
The current dialog is internal for our use, so in case you want something much more custom, we definitely suggest intercepting the long taps and showing your custom UI!

You could go over the code that displays our options and handles the actions and copy it over to your implementation if you need to, but as this is a very core/specific component, we don't allow overriding it completely within itself, we rather prefer having custom UI shown!
